components/chunk_text/fondant_component.yaml

```diff
@@ -0,0 +1,28 @@
+name: Chunk text
+description: |
+  Component that chunks text into smaller segments 
+  
+  This component takes a body of text and chunks into small chunks. The id of the returned dataset
+  consists of the id of the original document followed by the chunk index.
+
+image: fndnt/chunk_text:dev
+
+consumes:
+  text:
+    fields:
+      data:
+        type: string
+
+produces:
```

Comment:
What do you think about keeping a reference to the original document? E.g. doc_1 is chunked to doc_1_1, doc_1_2, ...
I can assume in some use case setup it is useful to find the original document of the chunk. 

We could also keep the text as it is and add an additional column `chunks` which will be a list of chunks.

Reply:
> What do you think about keeping a reference to the original document? E.g. doc_1 is chunked to doc_1_1, doc_1_2, ... I can assume in some use case setup it is useful to find the original document of the chunk.
> 
That's the currently the case since I am taking the original id of the document and appending the chunk number to it [here](https://github.com/ml6team/fondant/blob/d6b5480aba0489a160e35d7c71164d5cd29da645/components/chunk_text/src/main.py#L25)

> We could also keep the text as it is and add an additional column `chunks` which will be a list of chunks.

I think that's a good alternative, although just thinking about the next component (embedding) i'd rather have an embedding per row rather than a list of embeddings. I think it better matches also how things will be stored in the vector database.

Reply:
Agree with having one row per chunk. Adding `original_document_id` as a column makes sens I think though.

Reply:
yes might be better to link back to the original document, added!